compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/desugar/Desugar.java

```diff
@@ -2468,17 +2469,30 @@ private void createSimpleVarDefStmt(BLangSimpleVariable simpleVariable, BLangBlo
 
     @Override
     public void visit(BLangAssignment assignNode) {
-        boolean fieldAccessLVExpr = assignNode.varRef.getKind() == NodeKind.FIELD_BASED_ACCESS_EXPR;
+        boolean addNilToCastingType = shouldWidenExpressionTypeWithNil(assignNode);
         assignNode.varRef = rewriteExpr(assignNode.varRef);
         assignNode.expr = rewriteExpr(assignNode.expr);
         BType castingType = assignNode.varRef.getBType();
-        if (fieldAccessLVExpr) {
+        if (addNilToCastingType) {
             castingType = types.addNilForNillableAccessType(castingType);
         }
         assignNode.expr = types.addConversionExprIfRequired(rewriteExpr(assignNode.expr), castingType);
         result = assignNode;
     }
 
+    private static boolean shouldWidenExpressionTypeWithNil(BLangAssignment assignNode) {
+        BType exprType = assignNode.expr.getBType(); // This could be null for some std libs
```

Comment:
Do you have an example for this? I don't think it can be null, if it is, it may be due to a bug.

Reply:
create #41975